util/chplenv/third_party_utils.py

```diff
@@ -178,6 +178,25 @@ def pkgconfig_get_system_link_args(pkg, static=pkgconfig_default_static()):
     libs = libs_line.split()
     return ([ ], libs)
 
+#
+# Splits a linker argument string into a list of arguments. Args of the form
+# "-framework foo" must be treated as a single argument because of
+# de-duplication downstream, otherwise all but one "-framework" will be 
+# removed.
+#
+def parse_linker_args(str):
+    result = []
+    args = str.split()
+    i = 0
+    while (i < len(args)):
+        if args[i] == '-framework':
+            result.append(args[i] + " " + args[i+1])
+            i += 2
+        else:
+            result.append(args[i])
+            i += 1
+    return result
+
```

Comment:
This feels fragile. I'd prefer that we instead update the `dedup` to not remove `-framework` or maybe to only dedup `-l` args.

Reply:
Removed this code and modified `dedup` to only dedup `-l` args. That was the least complex solution.